airflow/opt/providers/etna/etna/etls/metis_linker.py

```diff
@@ -102,8 +102,13 @@ def data_frame_update(self, model_name, script, tail, update, metis, model):
                     else:
                         # Attempt to use pandas' automated detection
                         separator = None
-                # Maybe Future Feature: expose additional blanker value in config ui
-                data = pandas.read_table(file_reader, sep=separator, engine = 'python').replace({numpy.nan: None})
+                # Load dataframe
+                data = pandas.read_table(file_reader, sep=separator, engine = 'python', na_filter=False)
+                # Blank data equaling hole_value or empty string (None's ignored from the update-build later on)
+                replacements = {'': None}
+                if 'hole_value' in script:
```

Comment:
I think it is more explicit to have `data=data.replace(replacements)` nested under the if statement. It is also a bit unusual to set a Dict key as a empty string? Are you worried about bad input data? 

```python
if 'hole_value' in script:
    replacements = {script['hole_value']: None}
    data=data.replace(replacements)
```

Reply:
These seem like slightly different behaviors: the first is, always replace `''` with `None`, and also replace hole_value with `None`. The revision is only replace hole_value with `None`. However, the issue may be moot if pandas is already transforming `''` (i.e. what is between `,,` in a csv) into `None`

Reply:
> * I'm not sure `hole_value` is a very intuitive term for a user. Something like `value_to_skip`, `avoid_value`, `ignore_value`

Good point.  It was called `hole_value

> * I actually think it would be useful if the user could pass a comma separated list of values they want to ignore. Not sure if this is out of scope for a V1 though.
> * I am running into issues testing this locally, I've never linked a data frame before so it is probably something on my end. Getting: `MetisLoaderError('Invalid type for script data_frame for model population')`.

Reply:
> * I'm not sure `hole_value` is a very intuitive term for a user. Something like `value_to_skip`, `avoid_value`, `ignore_value`
> * I actually think it would be useful if the user could pass a comma separated list of values they want to ignore. Not sure if this is out of scope for a V1 though.

Fair points!  It was called `hole_value` in the etna gem version (`etna administrate models attributes update_from_csv __project_name__ __model_name__ __filepath__ --hole-value`), but I'm on board to give a better name, and like `values_to_skip`!  Multi-value was a thought I'd had at one point too.  Good poke -- I think I can code it easily enough -- and I like the comma separated list as the ui.

Reply:
> * I am running into issues testing this locally, I've never linked a data frame before so it is probably something on my end. Getting: `MetisLoaderError('Invalid type for script data_frame for model population')`.

Do you have an empty loader definition for the population model?  This seems like the catch for a barely initialized script: https://github.com/mountetna/monoetna/blob/6f6f89b17490d90bb6a97f87ab438ba2220c2fd1/airflow/opt/providers/etna/etna/etls/metis_linker.py#L187

Reply:
Re this and Saurabh's addition:
> I think it is more explicit to have `data=data.replace(replacements)` nested under the if statement. It is also a bit unusual to set a Dict key as a empty string? Are you worried about bad input data?

Originally, I was wanting to just ignore empty values, ie what is between `,,` in a csv as Saurabh put it.  AFAICT from the read_table docs, and now also from double-checking, emptiness is not ignored by default with a switch to having `na_filter=False` in the line 106 at the top of the preview here.

BUT re-thinking, the empty string is our blank value for string type data so always ignoring '' would block functionality of blanking string attributes from this path.

Testing of the behavior:
```
>>> df = pd.read_table('table.csv', sep = ',')
>>> df
          rec  parent comp
0  XX-YY1-ZZ1     NaN  abc
1  XX-YY1-ZZ2     NaN  abc
2  XX-YY1-ZZ3     NaN  abc
>>> df = pd.read_table('table.csv', sep = ',', na_filter=False)
>>> df
          rec parent comp
0  XX-YY1-ZZ1         abc
1  XX-YY1-ZZ2         abc
2  XX-YY1-ZZ3         abc
```

Reply:
Per this discussion:
- changed the name to values_to_ignore
- moved the `data.replace(replacements)` into the if statement as it will only be done if the config has this element. (a.k.a. empty strings are not always removed & allows for old configs that don't have this non-required element to still work)
- updated the usage to be a comma separated list
- added a display of the parsed targets where 'empty string' is used to stress that target =)

## New UI look
`,NA,NULL,<space>`
![image](https://github.com/mountetna/monoetna/assets/38870347/2b007908-2a32-4bb6-b04a-c6ef335a0356)

Reply:
oh my bad - I now see what your intention was. 

> BUT re-thinking, the empty string is our blank value for string type data so always ignoring '' would block functionality of blanking string attributes from this path.

Good point, if you wanted to making ignoring empty values optional, you could make a toggle?

Reply:
or empty string works in the description as you have! 

Was worried users would pass in a range of different inputs to mean `''` , but it is pretty explicit in your description

Reply:
> Was worried users would pass in a range of different inputs to mean '' , but it is pretty explicit in your description

Same!  Primarily came up with the targets display because I was also worried users might not understand either 1) how to target empty strings, or 2) that the empty field default translates to an empty string!